frontend/packages/console-dynamic-plugin-sdk/src/constants.ts

```diff
@@ -1,3 +1,36 @@
 export const extensionsFile = 'console-extensions.json';
 export const pluginManifestFile = 'plugin-manifest.json';
 export const remoteEntryFile = 'plugin-entry.js';
+export const featureReducerName = 'FLAGS';
+
+export enum FLAGS {
```

Comment:
I'm debating what to do with `FLAGS`. I feel like a more general API where you can test for a specific resource or access review without creating a flag would be better, but it doesn't cover all cases and it would be a large change.

I think if we expose `useFlag` we probably also need to expose this list. We should consider how we want to handle deprecated flags in the future.

Reply:
Flags are useful for enablement of extensions.
The same flags however are also useful in code. Specifically those which aren't resource based.

I don't know what most of these flags are used for :)
Perhaps we should do a review in our weekly meeting, before exporting them here.

Does Kubevirt need these flags right now?
If so you can just test for a string for now.

Reply:
Right, even for enablement of extensions, I wonder if we should generalize it to say enable if this resource is present rather than this flag. I'm thinking about environments like KCP where we can't assume any resource exists, even basic resources like pods. Does everything become a flag in that case? Are we worried about how that will scale?

I don't think we can solve that in this PR, but we might want to hold off making the enum API.

Reply:
@spadgett I'm all for omitting the enum from API for now. The flags API uses strings afterall.